### PR TITLE
Add the `wires` property to `catalyst.adjoint` and `catalyst.ctrl`

### PR DESCRIPTION
**Pull request by maliasadi**
Without implementing the `wires` property, the result of `catalyst.adjoint.wires` and `catalyst.ctrl.wires` would be `<Wires = [<WiresEnum.AnyWires: -1>]>`. This property is used in some of PennyLane's hybrid algorithms/demos. This PR adds this property to both of these Catalyst operations to unblock us with QJIT compiling those algorithms. 

[sc-55351]

### CODE REVIEW

File: frontend/catalyst/pennylane_extensions.py [hunk 1]
```diff
@@ -1239,6 +1247,17 @@ def decomposition(self):
...
 
+    @property
+    def wires(self):
+        """The list of all control-wires, work-wires, and active-wires."""
+
+        assert len(self.regions) == 1, "Qctrl is expected to have one region"
+        total_wires = sum(
+            (op.wires for op in self.regions[0].quantum_tape.operations),
+            self._control_wires + self._work_wires,
+        )
```
Comment (dime10) [inline frontend/catalyst/pennylane_extensions.py:1258]: In this scenario, are the control wires coming first?

File: frontend/test/pytest/test_adjoint.py [hunk 2]
```diff
@@ -463,5 +463,88 @@ def test_adjoint_outside_qjit():
...
+            )
+
+        qadj = C_adjoint(C_adjoint(C_adjoint(func)))()
+        return qadj.wires
+
+    # Without the `wires` property, returns `[-1]`
+    assert circuit() == qml.wires.Wires([0, 1])
+
+
+# # Variable wires is not supported!
```
Comment (dime10) [inline frontend/test/pytest/test_adjoint.py:510]: xfail?

File: frontend/catalyst/pennylane_extensions.py (hunk 1 above)
Comment (maliasadi) [reply frontend/catalyst/pennylane_extensions.py:1258]: yep, it's similar to PL

https://github.com/PennyLaneAI/pennylane/blob/343a6436bec71242ca20b3b28e24d1c83a8a99ed/pennylane/ops/op_math/controlled.py#L420

but I just realize that I should have added work_wires to the end of the list :)

Comment (dime10) [reply frontend/catalyst/pennylane_extensions.py]: Haha nice :D

File: doc/changelog.md [hunk 3]
```diff
@@ -13,6 +13,9 @@
 * Remove `qextract_p` and `qinst_p` from forced-order primitives.
   [(#469)](https://github.com/PennyLaneAI/catalyst/pull/469)
 
+* Add the `wires` property to `catalyst.adjoint` and `catalyst.ctrl`.
+  [(#480)](https://github.com/PennyLaneAI/catalyst/pull/480)
+
```
Comment (dime10) [inline doc/changelog.md:18]: Move this to bug fixes, and also state what type of scenario it fixes? :)

**Review by dime10 — APPROVED** [review]
Thanks @maliasadi, this is a good hotfix (and having the tests is actually great!) 💯

File: doc/changelog.md (hunk 3 above)
Comment (maliasadi) [reply doc/changelog.md:18]: good point! :+1:

**Pull request merged**